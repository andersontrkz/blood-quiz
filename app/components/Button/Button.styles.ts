import { StyleSheet } from 'react-native';
import { COLORS } from "@/constants";

const styles = StyleSheet.create({
  button: {
    marginTop: 20,
    width: '100%',
    backgroundColor: COLORS.primary,
    padding: 20,
    borderRadius: 5,
  },
  buttonText: {
    fontSize: 20,
    color: COLORS.white,
    textAlign: 'center',
    fontFamily: 'Poppins',
  },
});

export default styles;
