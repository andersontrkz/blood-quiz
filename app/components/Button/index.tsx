import React from 'react';
import { Text, TouchableOpacity, type TouchableOpacityProps } from 'react-native';
import styles from './Button.styles';

type ButtonProps = TouchableOpacityProps & {
    label: string;
};  

export default function Button({ label, ...rest }: ButtonProps) {
  return (
    <TouchableOpacity
      style={[styles.button, rest.style]}
      {...rest}
    >
      <Text style={styles.buttonText}>{ label }</Text>
    </TouchableOpacity>
  );
}
