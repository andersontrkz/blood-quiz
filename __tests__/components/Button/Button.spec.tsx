import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';

import Button from '@/app/components/Button';

describe('Button Component', () => {
  const onPressMock = jest.fn();
  const labelMock = 'Label';
  const accessibilityLabelMock = 'Accessibility Label';

  beforeEach(() => {
    onPressMock.mockReset();

    render(<Button label={labelMock} onPress={onPressMock} accessibilityLabel={accessibilityLabelMock} />);
  });

  it('should render correctly with the given label', () => {
    expect(screen.getByText(labelMock)).toBeTruthy();
  });

  it('should handle press events', () => {
    fireEvent.press(screen.getByText(labelMock));
    expect(onPressMock).toHaveBeenCalled();
  });

  it('should handle multiple press events', () => {
    fireEvent.press(screen.getByText(labelMock));
    fireEvent.press(screen.getByText(labelMock));
    expect(onPressMock).toHaveBeenCalledTimes(2);
  });

  it('should apply custom props correctly', () => {
    expect(screen.getByLabelText(accessibilityLabelMock)).toBeTruthy();
  });

  it('should be disabled when the disabled prop is passed', () => {
    render(<Button label={labelMock} onPress={onPressMock} disabled />);
    fireEvent.press(screen.getByText(labelMock));
    expect(onPressMock).not.toHaveBeenCalled();
  });

  it('should render without crashing if no onPress is provided', () => {
    render(<Button label={labelMock} />);
    fireEvent.press(screen.getByText(labelMock));
    expect(onPressMock).not.toHaveBeenCalled();
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<Button label={labelMock} onPress={onPressMock} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
