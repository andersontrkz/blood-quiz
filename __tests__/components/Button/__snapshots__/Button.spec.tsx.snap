// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#EE0F55",
      "borderRadius": 5,
      "marginTop": 20,
      "opacity": 1,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Poppins",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    Label
  </Text>
</View>
`;
